src/include/sof/audio/sink_api.h

```diff
@@ -160,4 +160,8 @@ int sink_set_alignment_constants(struct sof_sink __sparse_cache *sink,
 				 const uint32_t byte_align,
 				 const uint32_t frame_align_req);
 
+void sink_set_min_free(struct sof_sink __sparse_cache *sink, size_t min_free);
+size_t sink_get_min_free(struct sof_sink __sparse_cache *sink);
+bool sink_get_is_enough_free(struct sof_sink __sparse_cache *sink);
+
```

Comment:
these one line functions can be directly put to header file and with inline in declaration, see component_ext.h:
https://github.com/thesofproject/sof/blob/main/src/include/sof/audio/component_ext.h#L69

Reply:
No, I can't do it like this
Sink/source is a structure that **must not** be accessible directly (like private in C++). Therefore in sink_api.h the structure is not defined, all the user have is a forward declaration. 
Operations on the structure must be encapsulated in separate C file.